Engine/DataFeeds/Enumerators/Factories/DataQueueFuturesChainUniverseDataCollectionEnumerator.cs

```diff
@@ -0,0 +1,125 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Data;
+using QuantConnect.Data.UniverseSelection;
+using QuantConnect.Interfaces;
+using QuantConnect.Data.Market;
+using QuantConnect.Data.Auxiliary;
+using System.Collections;
+
+namespace QuantConnect.Lean.Engine.DataFeeds.Enumerators
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="ISubscriptionEnumeratorFactory"/> that reads
+    /// a live dataset provided by symbol universe provider into a <see cref="FuturesChainUniverseDataCollection"/> enumerator
+    /// </summary>
+    public class DataQueueFuturesChainUniverseDataCollectionEnumerator : IEnumerator<FuturesChainUniverseDataCollection>
+    {
+        private readonly List<BaseData> _symbolUniverse;
+        private readonly IEnumerator<BaseData> _underlying;
+        private readonly Symbol _symbol;
+        FuturesChainUniverseDataCollection _current;
+        bool _needNewCurrent;
```

Comment:
Style - I tend to prefer always being explicit about member accessibility modifiers.